### PR TITLE
Remove HeadSHA parameter from UpdateCheckRunOptions

### LINKED ISSUE #1652: Update Check Run HeadSHA
https://github.com/google/go-github/blob/1ff2714d8affd5ede5366bdb1a2d9472b669fe77/github/checks.go#L175-L186

Despite having an optional `HeadSHA` field, if you update this to a different commit hash it won't be updated.

We have some Check Runs that are PR-oriented and not commit-oriented and we were trying to use this field to rebind the Check Runs to the head commit of a PR on `synchronize`.

Is this a bug?

### PR DESCRIPTION
**Pull request by ishanupadhyay**
This PR is regarding [issue 1652.](https://github.com/google/go-github/issues/1652) head_sha is no longer supported as a parameter in case of [updating a check run](https://docs.github.com/en/free-pro-team@latest/rest/reference/checks#update-a-check-run). Therefore it is being removed from **type UpdateCheckRunOptions** in this PR. 

Fixes #1652.

### CODE REVIEW

**Comment by ishanupadhyay** [comment]
@gmlewis can you please review this PR ?

**Comment by gmlewis** [comment]
Note that this PR is a breaking API change, so I've added the label to keep track.

**Review by gmlewis — APPROVED** [review]
Thank you, @ishanupadhyay!
LGTM.

Awaiting second LGTM before merging.

**Comment by gmlewis** [comment]
@erikkn - it would be nice to get all breaking changes into the next major release (`v33.0.0`) so that we don't have to bump the major version more often than necessary.  If you feel like reviewing this PR, that would be greatly appreciated.

**Review by erikkn — APPROVED** [review]
We are aware of the breaking API change, so LGTM.

**Comment by gmlewis** [comment]
Thank you, @erikkn ! 
Merging.

**Pull request merged**
